docs/cow-protocol/tutorials/cow-swap/flash-loans.mdx

```diff
@@ -0,0 +1,264 @@
+* Some non-zero amount of the order's buy token in the Safe's balance, to pass CoW Swap order validation.
+
+The order itself intends to:
+1. Request a flash loan to Aave of 5,000 USDT.
+2. Use those 5,000 to partially repay the 10,000 USDT debt on Aave.
+3. Withdraw 400,000 USDC from Aave.
+4. Perform a trade on CoW Swap, selling up to 400,000 USDC for 5,002.5 USDT (borrowed amount + [0.05% flash loan fee on Aave](https://aave.com/docs/concepts/flash-loans)).
+2. Use the 5,002.5 USDT from the trade to fully repay the flash loan (including the fee).
+
+:::info
+
+Exchange ratios on the Sepolia network can be very inconsistent. To ensure successful execution of the order, a large amount of USDC is withdrawn from Aave and used in the trade.
+
+General considerations when testing flash loans on Sepolia and Aave:
+- [Token addresses in Aave may not correspond](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#2-select-aave-tokens) with tokens names in other dapps.
+- [Token ratios](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#4-determine-token-amounts-needed-to-trade-and-deposit) may need to be balanced, and trading amounts adjusted accordingly.
+- Tokens in Aave can be [obtained from the faucet](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#5-get-tokens-from-the-faucet).
+- Check [withdrawal and borrowing availability](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#7-check-aaves-withdrawalborrow-availability) of the tokens on Aave.
+
+In a production network (e.g. mainnet), these considerations generally won’t apply.
+
+:::
+
+## Aave repay pre-hook
+We need to define a [pre-hook](/cow-protocol/reference/core/intents/hooks) to repay the 5,000 USDT debt on Aave.
+
+As part of the settlement, we want to call the [`repay` function on Aave's pool](https://aave.com/docs/developers/smart-contracts/pool#write-methods-repay):
+```solidity
+function repay(
+    address asset,
+    uint256 amount,
+    uint256 interestRateMode,
+    address onBehalfOf
+) public virtual override returns (uint256)
+```
+
+The following Typescript code generates the corresponding function data using [`viem`](https://viem.sh/):
+```typescript
+const repayTxData = encodeFunctionData({
+    abi, // ABI definition for the repay function
+    functionName: 'repay',
+    args: [
+      '0xaA8E23Fb1079EA71e0a56F48a2aA51851D8433D0', // asset: USDT
+      '5000000000', // amount: 5,000 USDT (USDT has 6 decimals)
+      2, // interestRateMode: 2 for variable
+      '0x35eD9A9D1122A1544e031Cc92fCC7eA599e28D9C', // onBehalfOf: Safe address
+    ]
+  });
+```
+
+The `repay` call must be executed during the order settlement process. To enable this, the pre-hook defines a transaction that will be sent to the Safe wallet. The Safe then executes the actual repay call to the Aave pool.
```

Comment:
More specifically `repay` and `withdraw` absolutely have to be executed as a pre-hook because the settlement contract requires the user to have the collateral tokens before it pulls in the sell tokens of all settled orders.